### DISCUSSION TITLE
Adding to past commits

### DISCUSSION THREAD

**Discussion by donno2048** [community/community#22696] [New to GitHub]
<p>Consider the following scenario:</p>
<p>I have a project where all the past commits in it are fixup commits, now I’m starting to add things, so I want to add a <code>fixup</code> tag to the past commits.</p>
<p>How can I do it?</p>

**Comment by lee-dohm** [comment]
<p>Hello and welcome to the community!</p>
<p>I’m not sure I understand. Creating a fixup commit is one step in a two-step process: creating the fixup commit and rebasing the branch to squash the fixup commit and the fixup target. So are you saying that all previous commits in the repository were fixup commits that were never rebased? Or are you saying that you want to treat all previous commits in the repository as fixup commits so they can all be squashed into one commit?</p>

**Comment by donno2048** [comment]
<p>Thanks for the fast reaction,</p>
<p>I want to change all the previous commits to allow easier reading,</p>
<p>for example a commit like that: <code>commit number one</code></p>
<p>will be changed to <code>commit number one - fixup</code></p>
<p>is it possible?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If what you’re looking to do is to add a single word to every commit message in the entire repo, then yes, there are tools for that. The one built in to git is the <a href="https://git-scm.com/docs/git-filter-branch"><code>git filter-branch</code> command</a> which allows you to rewrite the history of an entire branch. You would use the <a href="https://git-scm.com/docs/git-filter-branch#Documentation/git-filter-branch.txt---msg-filterltcommandgt"><code>--msg-filter</code> option</a> to rewrite the commit messages.</p>
<p>I hope that helps!</p>
